src/Microsoft.Health.Fhir.Core/Features/Search/Expressions/ExpressionRewriter.cs

```diff
@@ -38,7 +38,16 @@ public virtual Expression VisitChained(ChainedExpression expression, TContext co
                 return expression;
             }
 
-            return new ChainedExpression(resourceTypes: expression.ResourceTypes, referenceSearchParameter: expression.ReferenceSearchParameter, targetResourceTypes: expression.TargetResourceTypes, reversed: expression.Reversed, expression: visitedExpression);
+            /*SortRewriter will return null expression if it finds _sort param same as on of filter params.
```

Comment:
> SortRewriter will return null expression if it finds _sort param same as one of filter params. But in case of chained names we do need to keep it.

Is it true for all the scenarios? Can it cause any regression?
Is there any other scenario where the visitedExpression is null, and it must return a null expression?

Reply:
look into ChainedExpression class line #38. It will return error if expression being passed is null. 
Only visitor other than SortRewriter that returns null is RemoveIncludesRewriter.  I have tested samples of requests where RemoveIncludesRewriter is called and it doesnt go through ChainedExpression. So from what I understand, this change shouldn't cause any regression.